core/base/src/TApplication.cxx

```diff
@@ -468,7 +468,46 @@ void TApplication::GetOptions(Int_t *argc, char **argv)
          } else {
             Warning("GetOptions", "-e must be followed by an expression.");
          }
+      } else if (!strcmp(argv[i], "--")) {
+         TObjString* macro = NULL;
 
+         if (fFiles) {
+            for (int f = 0; f < fFiles->GetEntriesFast(); f++) {
+               TObjString* file = dynamic_cast<TObjString*>(fFiles->At(f));
+
+               if (file->TestBit(kExpression))
+                  continue;
+               if (file->String().EndsWith(".root"))
+                  continue;
+               if (strchr(file->String().Data(), '('))
+                  continue;
+
+               if (macro)
+                  Warning("GetOptions", "-- is used with several macroses. "
+                                        "The arguments will be passed to the last one.");
+
+               macro = file;
+            }
+         }
+
+         if (macro) {
+            argv[i] = null;
+            ++i;
+            TString& str = macro->String();
+
+            str += '(';
+            for (; i < *argc; i++) {
+               str += argv[i];
+               str += ',';
+               argv[i] = null;
+            }
+            str.EndsWith(",") ? str[str.Length()-1] = ')' : str += ')';
```

Comment:
Except for adding spaces around the "-" in `Length() - 1`, I don't think clang-tidy's suggestions are of much value. So please ignore its "failure".